File: frontend/src/types/camera.ts
export enum CameraStatus {
  ONLINE = 'ONLINE',
  OFFLINE = 'OFFLINE',
  RECORDING = 'RECORDING',
  ERROR = 'ERROR',
}

export interface Camera {
  id: string;
  cameraCode?: string;
  name?: string;
  ipAddress?: string;
  location?: string;
  status?: CameraStatus | string;
}

File: gian_lan/src/main/java/com/idai/gian_lan/entity/Camera.java
package com.idai.gian_lan.entity;

import com.idai.gian_lan.dto.enums.CameraStatus;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Entity
@Table(name = "cameras")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Camera {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;

    @jakarta.persistence.Column(name = "camera_code", unique = true)
    String cameraCode;

    String name;

    String ipAddress;

    String location;

    @Enumerated(EnumType.STRING)
    CameraStatus status;
}
